exporter/collector/googlemanagedprometheus/monitoredresource.go

```diff
@@ -48,6 +48,28 @@ var promTargetKeys = map[string][]string{
 	instanceLabel:         {instanceLabel, semconv.AttributeServiceInstanceID, semconv.AttributeFaaSInstance},
 }
 
+// According to Cloud Monitoring docs, there are special values for
+// cluser in some runtimes.
+// See: https://cloud.google.com/stackdriver/docs/managed-prometheus/setup-opsagent
+func (c Config) getClusterNameByEnvironment(res pcommon.Resource) string {
+	attrs := res.Attributes()
+
+	if !c.UseEnvironmentClusterNames {
+		return getStringOrEmpty(attrs, promTargetKeys[clusterLabel]...)
+	}
+
+	cloudPlatform := getStringOrEmpty(attrs, semconv.AttributeCloudPlatform)
+	switch cloudPlatform {
+	case semconv.AttributeCloudPlatformGCPComputeEngine:
+		return "__gce__"
+	case semconv.AttributeCloudPlatformGCPCloudRun:
+		return "__run__"
+	default:
+		// No special rules, look up cluster as normal.
+		return getStringOrEmpty(attrs, promTargetKeys[clusterLabel]...)
```

Comment:
can this be the primary lookup?  i.e. if a user has set the `cluster` resource attribute manually, it would still respected.  Then we can fall back to the platform-based cluster names if the primary keys are not found.

Reply:
Done, did a slight refactor to the `getStringOr*` methods.